### PR TITLE
Flesh out base documentation with introduction/background/details

### PR DESCRIPTION
**Pull request by kratsg**
# Description

This PR fleshes out our documentation so it is a little less bare-bones and gives people things to read and understand the history/background behind the project.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add background information on HistFactory
* Detail some of the mathematical formulation used in pyhf
* Add schema discussion in more detail
```

### CODE REVIEW

**Comment by lukasheinrich** [comment]
do we have a way to preview the docs of a PR? I guess not but it might be useful (not sure if there is such a functionality in RTD)

**Comment by kratsg** [comment]
> do we have a way to preview the docs of a PR? I guess not but it might be useful (not sure if there is such a functionality in RTD)

could develop staging with GH actions.

**Comment by matthewfeickert** [comment]
> could develop staging with GH actions.

Can you elaborate? You're saying have some sort of separate deploy area for visual inspection? Or am I totally lost?

**Comment by kratsg** [comment]
> Can you elaborate? You're saying have some sort of separate deploy area for visual inspection? Or am I totally lost?

Basically, GH actions should let you deploy to some website on every PR (or at least, say, trigger some integratoin/webhook) and master is built with GH-pages.

**Comment by kratsg** [comment]
https://pyhf-staging.readthedocs.io/en/docs-pubdate/

**Comment by matthewfeickert** [comment]
@kratsg  First, the docs you made are freaking amazing. Really nice! :smiley:  I have 4 questions though:

1. Do you know how we can get the left LaTeX quotes to render correctly around "auxiliary measurements"? 
```latex
``auxiliary measurement''
```
Doesn't work
2. Can you remind @lukasheinrich  and me how you are actually getting https://pyhf-staging.readthedocs.io/en/docs-pubdate/ deployed? I don't see an added file in the PR that does this, so did you just set it up to watch? We're both just curious.
3. At the moment the text is split across lines as it wraps over around 80 spaces. Do you have any objection to me making it 1 sentence per line (now that you've done the hard work)?
4. You had brought up using netlify. Do you have any further thoughts on that here?

**Comment by kratsg** [comment]
1. It's fixed now (see https://mathhelpboards.com/math-formulas-mathjax-62/rh-lh-double-quotes-mathjax-1511.html). Note that this isn't latex, but MathJax.

Screenshot 2019-06-24 07 24 37

2. You can set up a RTD build without setting up the webhook. It just means you're doing (manual) p(o|u)lling instead of triggers.

3. We don't do that in the rest of the documentation. If there's no `black` equivalent here, I'd like to leave it as it is.

4. It would be nice to set it up if I had permissions to change the settings of this project... @lukasheinrich ?

**Comment by kratsg** [comment]
Builds can be (re)started manually here: https://readthedocs.org/projects/pyhf-staging/builds/

**Comment by matthewfeickert** [comment]
> 2. We don't do that in the rest of the documentation. If there's no `black` equivalent here, I'd like to leave it as it is.

Hm. I hadn't noticed this as I've mostly been writing docstrings with one sentence per line. (I've noticed there are some edge cases in which this matters, but I don't have an example I can point you to.)

Would you be open to the idea of at least starting a new line after each sentence? This makes it easier for me to focus on the sentence as an atomic chunck (but I also have softwrap on).

File: docs/intro.rst [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+   \label{eqn:sample_rates}
+    \nu_{cb}\left(\fullset\right) = \sum_{s\in\mathrm{\,samples}} \nu_{scb}\left(\freeset,\constrset\right) = \sum_{s\in\mathrm{\,samples}}\underbrace{\left(\prod_{\kappa\in\,\bm{\kappa}} \kappa_{scb}\left(\freeset,\constrset\right)\right)}_{\text{multiplicative modifiers}}\, \Bigg(\nu_{scb}^0\left(\freeset, \constrset\right) + \underbrace{\sum_{\Delta\in\bm{\Delta}} \Delta_{scb}\left(\freeset,\constrset\right)}_{\text{additive modifiers}}\Bigg)\,.
+
+The total rates are the sum over sample rates :math:`\nu_{csb}`, each
+determined from a *nominal rate* :math:`\nu_{scb}^0` and a set of multiplicative and
+additive denoted *rate modifiers* :math:`\bm{\kappa}(\fullset)` and
+:math:`\bm{\Delta}(\fullset)`. These modifiers are functions of (usually
+a single) model parameters. Starting from constant nominal rates, one
+can derive the per-bin event rate modification by iterating over all
+sample rate modifications as shown in .
```
Comment (matthewfeickert) [inline docs/intro.rst]: shown in where?

File: docs/intro.rst [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+equally.  For constrained modifiers, the implied constraint term is given as
+well as the necessary input data required to construct it.  :math:`\sigma_b`
+corresponds to the relative uncertainty of the event rate, whereas
+:math:`\delta_b` is the event rate uncertainty of the sample relative to the
+total event rate :math:`\nu_b = \sum_s = \nu^0_{sb}`.
+
+Modifiers implementing uncertainties are paired with
+a corresponding default constraint term on the parameter limiting the
+rate modification. The available modifiers may affect only the total
+number of exected events of a sample within a given channel, i.e. only
```
Comment (matthewfeickert) [inline docs/intro.rst]: `exected` should be `expected`

Comment (matthewfeickert) [reply docs/intro.rst]: Resolved in https://github.com/diana-hep/pyhf/pull/480/commits/42c558a711fa1031ea8a3d13a7297375f4b72505

File: docs/intro.rst (hunk 1 above)
Comment (kratsg) [reply docs/intro.rst]: fixed in e930616

**Comment by kratsg** [comment]
> Would you be open to the idea of at least starting a new line after each sentence? This makes it easier for me to focus on the sentence as an atomic chunck (but I also have softwrap on).

Ok, but we really only have this week to get the docs out. We're on a short timeline.

**Pull request merged**